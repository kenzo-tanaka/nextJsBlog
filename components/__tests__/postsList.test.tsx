import React from "react";
import { createRenderer } from "react-test-renderer/shallow";
import PostsList from "../postsList";
import { PostData } from "@types";

const posts: PostData[] = [
  {
    slug: "test-slug",
    title: "Test title",
    date: "2021-04-17",
    content: "This is a test post",
    category: "work",
  },
];
const renderer = createRenderer();

test("PostsList", () => {
  renderer.render(<PostsList posts={posts} />);
  const renderedOutput = renderer.getRenderOutput();
  expect(renderedOutput).toMatchSnapshot();
});
