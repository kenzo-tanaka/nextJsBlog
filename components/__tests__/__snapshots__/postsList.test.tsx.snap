// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsList 1`] = `
<section>
  <ul>
    <li
      className="mb-6"
    >
      <Link
        href="/posts/test-slug"
      >
        <a
          className="text-lg font-semibold text-gray-800 visited:text-gray-600"
        >
          Test title
        </a>
      </Link>
      <br />
      <small
        className="text-sm font-normal text-gray-400"
      >
        <Date
          dateString="2021-04-17"
        />
      </small>
    </li>
  </ul>
</section>
`;
